server/courselist.rb

```diff
@@ -679,6 +681,13 @@ def self.getURLUniqname(url)
 
   #################### Data provider functions #################
 
+  # If a response response from the provider is throttled then log that fact.
+  SERVICE_UNAVAILABLE = "503"
+
+  def logIfUnavailable(response,msg)
+    logger.warn("provider response throttled or unavailable for: #{msg}") if SERVICE_UNAVAILABLE.casecmp(response.meta_status.to_s).zero?
```

Comment:
We don't need the casecmp but IMHO it's more efficient for developers to always put it in for string compare.  If nothing else this avoids having case bugs if the string, at some point, includes characters.  There may be circumstances when case is important for a compare but respecting case is more likely to be a problem than ignoring case.